core/workflow-operator/src/main/scala/edu/uci/ics/amber/operator/sink/managed/ProgressiveSinkOpExec.scala

```diff
@@ -2,15 +2,22 @@ package edu.uci.ics.amber.operator.sink.managed
 
 import edu.uci.ics.amber.core.executor.SinkOperatorExecutor
 import edu.uci.ics.amber.core.storage.model.BufferedItemWriter
+import edu.uci.ics.amber.core.storage.result.ResultStorage
 import edu.uci.ics.amber.core.tuple.{Tuple, TupleLike}
 import edu.uci.ics.amber.operator.sink.{IncrementalOutputMode, ProgressiveUtils}
+import edu.uci.ics.amber.virtualidentity.{OperatorIdentity, WorkflowIdentity}
 import edu.uci.ics.amber.workflow.PortIdentity
 
-class ProgressiveSinkOpExec(outputMode: IncrementalOutputMode, storage: BufferedItemWriter[Tuple])
-    extends SinkOperatorExecutor {
+class ProgressiveSinkOpExec(
+    outputMode: IncrementalOutputMode,
+    storageKey: String,
+    workflowIdentity: WorkflowIdentity
+) extends SinkOperatorExecutor {
+  val writer: BufferedItemWriter[Tuple] =
+    ResultStorage.getOpResultStorage(workflowIdentity).get(OperatorIdentity(storageKey)).writer()
```

Comment:
Since we are talking about using the PartitionDocument as one of the implementation of the storage. Each worker gonna write its own partition so when creating the writer, the worker ID must be passed in some way.

So is the worker id available here? If so we can probably use worker id as part of the storageKey. If not we need to have a design for that.

Reply:
Following up on the PartitionDoc, do we currently have a way to use a URI to create a writer? I'm a bit concerned that this PR could result in storage failures across a cluster, even for MongoDB, which already supports distributed partitioned collections.

Reply:
We should have such a way to create a writer/reader with a URI. but currently we don't have it.

Reply:
to be done in a future PR.